File: sdk/SDK.Examples/src/CreatePackageWithSMSAuthentication.cs
using System;
using Silanis.ESL.SDK;
using Silanis.ESL.SDK.Builder;

namespace SDK.Examples
{
	public class CreatePackageWithSMSAuthentication
	{
		public static string apiToken = "YOUR TOKEN HERE";
		public static string baseUrl = "ENVIRONMENT URL HERE";

		public static void Main (string[] args)
		{
			// Create new esl client with api token and base url
			EslClient client = new EslClient (apiToken, baseUrl);

			//Signer firstname, lastname, email, title, company
			DocumentPackage package = PackageBuilder.NewPackageNamed ("C# Package " + DateTime.Now)
					.DescribedAs ("This is a new package")
					.ExpiresOn(DateTime.Now.AddDays(5))
					.WithEmailMessage("This message should appear in email invitation to signers")
					.InPerson(true)
					.WithSigner(SignerBuilder.NewSignerWithEmail("john.smith@email.com")
					            .WithFirstName("John")
					            .WithLastName("Smith")
					            .WithTitle ("Super Duper")
					            .WithCompany ("Acme Inc")
					            .WithSMSSentTo("3334445555"))
					.Build ();

			PackageId id = client.CreatePackage (package);

			Console.WriteLine ("Package {0} was created", id.Id);
		}
	}
}

File: sdk/SDK.VBExamples/SDK.VBExamples/CreatePackageWithSMSAuthentication.vb
Imports System
Imports Silanis.ESL.SDK
Imports Silanis.ESL.SDK.Builder

Namespace SDK.VBExamples
    Public Module CreatePackageWithSMSAuthentication
        Dim apiToken = "YOUR TOKEN HERE"
        Dim baseUrl = "ENVIRONMENT URL HERE"

        Sub Main()
            ' Create new esl client with api token and base url
            Dim client = New EslClient(apiToken, baseUrl)

            'Signer firstname, lastname, email, title, company
            Dim package = PackageBuilder.NewPackageNamed("VB Package " + DateTime.Now) _
                .DescribedAs("This is a new package") _
                .ExpiresOn(DateTime.Now.AddDays(5)) _
                .WithEmailMessage("This message should appear in email invitation to signers") _
                .InPerson(True) _
                .WithSigner(SignerBuilder.NewSignerWithEmail("john.smith@email.com") _
                    .WithFirstName("John") _
                    .WithLastName("Smith") _
                    .WithTitle("Super Duper") _
                    .WithCompany("Acme Inc") _
                    .WithSMSSentTo("3334445555")) _
                .Build()

            Dim id = client.CreatePackage(package)

            Console.WriteLine("Package {0} was created", id.Id)
        End Sub
    End Module
End Namespace